File: src/main/java/com/matera/restserver/repository/EmployeeRepository.java
package com.matera.restserver.repository;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

import com.matera.restserver.model.Employee;

public interface EmployeeRepository extends JpaRepository<Employee, Long> {

	List<Employee> findByIdAndStatus(Long id, String status);

	List<Employee> findByStatus(String status);

}

File: src/main/kotlin/com/matera/restserver/repository/EmployeeRepository.kt
package com.matera.restserver.repository

import com.matera.restserver.model.Employee
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface EmployeeRepository : JpaRepository<Employee, UUID> {
    fun findByIdAndStatus(id: UUID, status: String): Employee?
    fun findByStatus(status: String): List<Employee>
}
